### ISSUE TITLE
Get daily and hourly forecast for single station.

### ISSUE THREAD

**Issue by Gibby** [briis/smartweather#53]
I'd like to be able to pull both Daily and Hourly forecast for a single station. 

I put in a quick little dirty change in my own fork that appends the forecast type to the name of the entity.

**Note:** Needs work, since its possible to pull on sensors still, but it would be nice to be supported here. So I only pull all sensors for the hourly forecast and leave it unchecked for the daily.

I am 100% sure there are better ways to do this, maybe instead of a drop down, just a check box for Daily and Hourly.

```diff --git a/custom_components/smartweather/config_flow.py b/custom_components/smartweather/config_flow.py
index 2e5e067..c61555a 100644
--- a/custom_components/smartweather/config_flow.py
+++ b/custom_components/smartweather/config_flow.py
@@ -87,9 +87,6 @@ class SmartWeatherFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
             return await self._show_setup_form(errors)
 
         entries = self._async_current_entries()
-        for entry in entries:
-            if entry.data[CONF_ID] == unique_id:
-                return self.async_abort(reason="station_exists")
 
         return self.async_create_entry(
             title=unique_id,
diff --git a/custom_components/smartweather/entity.py b/custom_components/smartweather/entity.py
index 0939ad6..ffa053b 100644
--- a/custom_components/smartweather/entity.py
+++ b/custom_components/smartweather/entity.py
@@ -10,6 +10,7 @@ from .const import (
     DEFAULT_BRAND,
     DEFAULT_ATTRIBUTION,
     DEVICE_TYPE_WEATHER,
+    CONF_FORECAST_TYPE,
 )
 
 
@@ -31,8 +32,9 @@ class SmartWeatherEntity(Entity):
         self._platform_serial = self.server["serial_number"]
         self._platform_id = server["station_type"]
         self._device_key = f"{self.entries[CONF_STATION_ID]}"
+        self._forecast_type = f"{self.entries[CONF_FORECAST_TYPE]}"
         if self._entity == DEVICE_TYPE_WEATHER:
-            self._unique_id = self._device_key
+            self._unique_id = f"{self._device_key}_{self._forecast_type}"
         else:
             self._unique_id = f"{self._device_key}_{self._entity}"
 
diff --git a/custom_components/smartweather/manifest.json b/custom_components/smartweather/manifest.json
index cdc7058..3307a2c 100644
--- a/custom_components/smartweather/manifest.json
+++ b/custom_components/smartweather/manifest.json
@@ -9,6 +9,6 @@
         "@briis"
     ],
     "requirements": [
-        "pysmartweatherio==2.1.5"
+        "pysmartweatherio==2.1.4"
     ]
 }
\ No newline at end of file
diff --git a/custom_components/smartweather/weather.py b/custom_components/smartweather/weather.py
index ee432b8..cbf132f 100644
--- a/custom_components/smartweather/weather.py
+++ b/custom_components/smartweather/weather.py
@@ -96,9 +96,9 @@ class SmartWeatherWeather(SmartWeatherEntity, WeatherEntity):
         super().__init__(
             coordinator, entries, device_type, server, fcst_coordinator, None
         )
-        self._name = f"{DOMAIN.capitalize()} {entries[CONF_ID]}"
-        self._unit_system = unit_system
         self._forecast_type = fcst_type
+        self._name = f"{DOMAIN.capitalize()} {entries[CONF_ID]} {self._forecast_type}"
+        self._unit_system = unit_system
 
     @property
     def name(self) -> str:
```

**Comment by briis** [comment]
Thank you for this. Do you have a working version, or is this just an example on how you would separate the the two forecast types? 

Per design, the Weather Entity is not built to cope with both daily and hourly data at the same time, so I see some issues there.

**Comment by Gibby** [comment]
@briis the above is what I am using in my HA so I can get both the Daily and Hourly forecast details.

**Comment by briis** [comment]
Hi @Gibby 
I have now made some modifications that enables this feature. Please have a look at my comment here: https://github.com/briis/smartweather/issues/51#issuecomment-768964768

**Comment by briis** [comment]
I have released v2.2.2 that includes this, so I will close this issue.

**Comment by Gibby** [comment]
@briis sorry for just getting to this, working great!
